packages/amplify-provider-awscloudformation/src/aws-utils/aws-cfn.js

```diff
@@ -73,99 +82,115 @@ class CloudFormation {
     });
   }
 
-  readStackEvents(stackName) {
-    // Stores all the events displayed to the user (useful to do a diff with new Events)
-    let allShownEvents = [];
-    const self = this;
-    const cfnModel = this.cfn;
-    const params = {
-      StackName: stackName,
-    };
-    let nestedStackQueue = [];
-
-    this.pollForEvents = setInterval(() => {
-      cfnModel.describeStackEvents(params, (err, data) => {
-        if (err) {
-          console.log(err);
-          /* Not throwing error out here since this is an asynchornous event
-          running in the background */
-        } else {
-          let stackEvents = data.StackEvents;
-          stackEvents = stackEvents
-            .filter(event => self.eventStartTime < new Date(event.Timestamp));
-
-          if (allShownEvents.length === 0) {
-            // Showing events for the first time to the user
-            if (stackEvents.length > 0) {
-              showEvents(stackEvents);
-              allShownEvents = stackEvents;
-            }
-          }
+  collectStackErrors(stackName) {
+    // add root stack to see the new stacks
+    this.readStackEvents(stackName);
+    // wait for the poll queue to drain
+    return new Promise((resolve) => {
+      this.pollQueue.once('empty', () => {
+        const failedStacks = this.stackEvents.filter(ev =>
+          CNF_ERROR_STATUS.includes(ev.ResourceStatus));
+        try {
+          const trace = this.generateFailedStackErrorMsgs(failedStacks);
+          console.log(`\n\n${chalk.reset.red.bold('Following resources failed')}\n`);
+          trace.forEach((t) => {
+            console.log(t);
+            console.log('\n');
+          });
+          resolve();
+        } catch (e) {
+          Promise.reject(e);
+        }
+      });
+    });
+  }
 
-          let newEvents = stackEvents;
-          const nestedStackEventPromises = [];
+  generateFailedStackErrorMsgs(eventsWithFailure) {
+    const { envName } = this.context.amplify.getEnvInfo();
+    const stackTrees = eventsWithFailure
+      .filter(stack => stack.ResourceType !== 'AWS::CloudFormation::Stack')
+      .map((event) => {
+        const err = [];
+        const resourceName = event.PhysicalResourceId.replace(`-${envName}`, '') || event.LogicalResourceId;
```

Comment:
why do we tamper with the actual PhysicalResourceId?

Reply:
We don't . Strings are immutable

Reply:
So why are we replaceing envName with ''?

Reply:
So when we print the resource name, it won't have `-env` appended to it

Reply:
What if the envName is NONE? and for Lex it is not `-envName` and it is `_envName`. i think we should avoid doing this replace operation.

Reply:
It doesn't harm, if there is no env name

Reply:
I can think about cases (although edge cases) where this could show inconsistent results, one of them would be in case of lex and then one more that I could think of is when the resource name itself has a `-envName` in it's name.

Reply:
Are there any other patterns? Or is it just `-` and `_`

Reply:
From the top of my head, I can think of these two but there could me more in the future

Reply:
Accommodated both cases